lgsm/functions/command_monitor.sh

```diff
@@ -11,76 +11,79 @@ commandaction="Monitoring"
+	elif [[ "$(head -n 1 "${lockdir}/${selfname}.lock")" =~ [A-Za-z] ]]; then
+		fn_print_warn_nl "Checking lockfile: fixing illegal lockfile"
 		date '+%s' > "${lockdir}/${selfname}.lock"
 		echo "${version}" >> "${lockdir}/${selfname}.lock"
 		echo "${port}" >> "${lockdir}/${selfname}.lock"
+	else
+		fn_print_ok_nl "Checking lockfile"
 	fi
 }
 
 fn_monitor_check_update(){
+	fn_print_dots "Checking active updates"
+
 	# Monitor will check if update is already running.
 	if [ "$(pgrep "${selfname} update" | wc -l)" != "0" ]; then
-		fn_print_dots "Checking active updates: "
-		fn_print_checking_eol
-		fn_script_log_info "Checking active updates: CHECKING"
-		fn_print_error_nl "Checking active updates: SteamCMD is currently checking for updates: "
-		fn_print_error_eol
-		fn_script_log_error "Checking active updates: SteamCMD is currently checking for updates: ERROR"
+		fn_print_fail_nl "SteamCMD is currently checking for updates"
+		exitcode="2"
 		core_exit.sh
+	else
+		fn_print_ok_nl "Checking active updates"
 	fi
 }
 
-fn_monitor_check_session(){
-	fn_print_dots "Checking session: "
-	fn_print_checking_eol
-	fn_script_log_info "Checking session: CHECKING"
+fn_monitor__is_server_running(){
+	fn_print_dots "Checking session"
+
 	# uses status var from check_status.sh
 	if [ "${status}" != "0" ]; then
-		fn_print_ok "Checking session: "
-		fn_print_ok_eol_nl
-		fn_script_log_pass "Checking session: OK"
+		fn_print_ok_nl "Checking session"
+		return 0
 	else
-		fn_print_error "Checking session: "
-		fn_print_fail_eol_nl
-		fn_script_log_fatal "Checking session: FAIL"
-		alert="restart"
-		alert.sh
-		fn_script_log_info "Checking session: Monitor is restarting ${selfname}"
-		command_restart.sh
-		core_exit.sh
+		fn_print_error_nl "Checking session"
+		return 1
 	fi
 }
 
-fn_monitor_check_queryport(){
-	# Monitor will check queryport is set before continuing.
-	if [ -z "${queryport}" ]||[ "${queryport}" == "0" ]; then
-		fn_print_dots "Checking port: "
-		fn_print_checking_eol
-		fn_script_log_info "Checking port: CHECKING"
-		if [ -n "${rconenabled}" ]&&[ "${rconenabled}" != "true" ]&&[ ${shortname} == "av" ]; then
-			fn_print_warn "Checking port: Unable to query, rcon is not enabled"
-			fn_script_log_warn "Checking port: Unable to query, rcon is not enabled"
-		else
-			fn_print_error "Checking port: Unable to query, queryport is not set"
-			fn_script_log_error "Checking port: Unable to query, queryport is not set"
-		fi
-		core_exit.sh
+fn_monitor__is_queryport_valid(){
+	fn_print_dots "Checking port value: \"${queryport}\""
+
+	if ! grep -qe '^[1-9][0-9]*$' <<< "${queryport}"; then
+		fn_print_error_nl "Checking port value: \"${queryport}\" - Unable to query, queryport is illegal \"${queryport}\" and rcon=\"${rconenabled}\". This can be fine if server didn't already create the config file so we couldn't extract information from it, try to rerun."
```

Comment:
rconenabled is not available for most servers so mostly won't be relevant.

Reply:
The output could depend if its set but than its maybe unset and not printed where it should be set and there would be no output. Remove or improve?